src/leo_cache_tran.erl

```diff
@@ -0,0 +1,217 @@
+
+
+%%--------------------------------------------------------------------
+%% GEN_SERVER CALLBACKS
+%%--------------------------------------------------------------------
+%% Function: init(Args) -> {ok, State}          |
+%%                         {ok, State, Timeout} |
+%%                         ignore               |
+%%                         {stop, Reason}
+%% Description: Initiates the server
+init([]) ->
+    case ets:new(?PROCESSDB, [named_table, set, private]) of
+        ?PROCESSDB ->
+            ok;
+        _ ->
+            erlang:error({error, could_not_create_ets_table})
+    end,
+    case ets:new(?REPLYDB, [named_table, set, private]) of
+        ?REPLYDB ->
+            ok;
+        _ ->
+            erlang:error({error, could_not_create_ets_table})
+    end,
+    {ok, unused, ?TRAN_TIMEOUT}.
+
+handle_call(stop, _From, State) ->
+    {stop, normal, ok, State};
+
+%%handle_call({unregister, Tbl, Key}, From, State) ->
+%%    {Pid, _Ref} = From,
+%%    case ets:lookup(?REPLYDB, {Tbl, Key}) of
+%%        [{{Tbl, Key}, ReplyList}] ->
+%%            ReplyList2 = lists:filter(fun({Pid2, _Ref2}) ->
+%%                                              Pid2 =/= Pid
+%%                                      end, ReplyList),
+%%            true = ets:insert(?REPLYDB, {{Tbl, Key}, ReplyList2});
+%%        _ ->
+%%            void
+%%    end,
+%%    {reply, ok, State, ?TRAN_TIMEOUT};
+
+handle_call({begin_tran, Pid, Tbl, Key}, _From, State) ->
+    MonitorRef = erlang:monitor(process, Pid),
+    true = ets:insert(?PROCESSDB, {MonitorRef, {Tbl, Key}}), 
+    case ets:insert_new(?REPLYDB, {{Tbl, Key}, []}) of
+        true ->
+            {reply, ok, State, ?TRAN_TIMEOUT};
+        false ->
+            {reply, {error, in_process}, State, ?TRAN_TIMEOUT}
+    end;
+
+handle_call({wait_tran, Tbl, Key}, From, State) ->
+    case ets:lookup(?REPLYDB, {Tbl, Key}) of
+        [{{Tbl, Key}, ReplyList}] ->
+            true = ets:insert(?REPLYDB, {{Tbl, Key}, [From | ReplyList]}),
+            {noreply, State, ?TRAN_TIMEOUT};
+        _ ->
+            {reply, {ok, not_found}, State, ?TRAN_TIMEOUT}
+    end;
+
+handle_call({end_tran, Tbl, Key}, _From, State) ->
+    reply_all(Tbl, Key),
+    {reply, ok, State, ?TRAN_TIMEOUT}.
+
+handle_cast({end_tran, Tbl, Key}, State) ->
+    reply_all(Tbl, Key),
+    {noreply, State};
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+
+%% Function: handle_info(Info, State) -> {noreply, State}          |
+%%                                       {noreply, State, Timeout} |
+%%                                       {stop, Reason, State}
+%% Description: Handling all non call/cast messages
+%% handle_info({_Label, {_From, MRef}, get_modules}, State) ->
+%%     {noreply, State};
+handle_info({'DOWN', MonitorRef, _Type, _Pid,_Info}, State) ->
+    case ets:lookup(?PROCESSDB, MonitorRef) of
+        [{MonitorRef, {Tbl, Key}}] ->
+            reply_all(Tbl, Key);
+        _ ->
+            void
+    end,
```

Comment:
Add document for developers using this module to nofify that a caller process must not be pooled.
in other words, to be temporal process like leo_large_object_get_handler.